java/registry/src/main/java/dev/sunbirdrc/registry/service/ElasticSearchService.java

```diff
@@ -101,7 +119,88 @@ public JsonNode search(JsonNode inputQueryNode) throws IOException {
 
     }
 
-	private void updateStatusFilter(SearchQuery searchQuery) {		
+    private ArrayNode expandReference(JsonNode searchedNode) {
+        ArrayNode arrayNode = (ArrayNode) searchedNode;
+        ArrayNode nodeWithExpandedReference = JsonNodeFactory.instance.arrayNode();
+        HashMap<String, List<String>> indexOsidsMap = new HashMap<>();
+        for (JsonNode node : arrayNode) {
+            ObjectNode objectNode = (ObjectNode) node;
+            objectNode.fields().forEachRemaining(objectField -> {
+                if(objectField.getValue().asText().startsWith("did:")) {
```

Comment:
same here as well

Reply:
Added check based on the regex for DID format. Made the same changes here as well